### PR TITLE
WIP - Adds some tests for tests with duplicate ids and includes real …

### PR DESCRIPTION
**Pull request by samholder**
…sample file of the issue from the wild

This PR needs more unit tests to verify the updating behaviour, and needs the integration test to be removed (this was only added to verify that the real files can be loaded)

### CODE REVIEW

File: test/dotnet-trx-merge.IntegrationTests/DotNetTrxMergeTests.cs [hunk 1]
```diff
@@ -112,8 +112,28 @@ public async Task DotnetTrxMerge_WithOutputFolder_Success()
         text.Should().Contain("testId=\"86e2b6e4-df7a-e4fa-006e-c056c908e219\"");
         text.Should().Contain("testName=\"SecondSimpleNumberCompare\"");
     }
-    
-    
+
+    [Fact]
+    public async Task DotnetTrxMerge_WithMultipleFilesFromOurTestRun()
```
Comment (samholder) [inline test/dotnet-trx-merge.IntegrationTests/DotNetTrxMergeTests.cs:117]: this test fails, and is not needed, I was just using it to verify that my changes resulted in being able to load the files

File: test/dotnet-trx-merge.IntegrationTests/DotNetTrxMergeTests.cs [hunk 2]
```diff
@@ -40,7 +40,7 @@ public void DotnetTrxMerge_WithSelectingFiles_Success()
         output.Should().Contain("Found 2 files to merge");
         output.Should().Contain("Found 1 tests in file");
         output.Should().Contain("Found 2 tests in file");
-        output.Should().Contain("New result of test 3dacbae9-707e-1881-d63c-3573123dffc6 was found in file ");
+        output.Should().Contain("New result of test 3dacbae9-707e-1881-d63c-3573123dffc6 - 631e6252-66d2-49b8-9fb3-6b9fc02425db was found in file ");
```
Comment (samholder) [inline test/dotnet-trx-merge.IntegrationTests/DotNetTrxMergeTests.cs:43]: should maybe consider testing no other 'new result of test' things in here, as we expect only one change

File: test/dotnet-trx-merge.IntegrationTests/Fixtures/MergeWithMyFiles/_fv-az635-124_2023-10-04_11_32_15.trx [hunk 3]
```diff
@@ -0,0 +1,1408 @@
...
+    <UnitTestResult executionId="c8c636c5-8d1a-40d1-9b20-65d968042613" testId="7f021ba4-ce47-c759-0e9a-e85913d720db" testName="ShouldReturnCorrectTransactionStatus (BLOCKED,CompleteFailed,Optimis.SmartPayBureau.Client.error[])" computerName="fv-az635-124" duration="00:00:00.0000078" startTime="2023-10-04T11:32:16.6593360+00:00" endTime="2023-10-04T11:32:16.6600670+00:00" testType="13cdc9d9-ddb5-4fa4-a97d-d965ccfc6d4b" outcome="Passed" testListId="8c84fa94-04c1-424b-9868-57a2d4851a1d" relativeResultsDirectory="c8c636c5-8d1a-40d1-9b20-65d968042613" />
+    <UnitTestResult executionId="868217d3-4a8f-4036-9886-314c10b3f4b9" testId="bd0f3e87-2d49-7223-2c00-e0924f9c83f5" testName="SubsequentRecurringCardNoUserDetail" computerName="fv-az635-124" duration="00:00:00.0006997" startTime="2023-10-04T11:32:16.3624740+00:00" endTime="2023-10-04T11:32:16.4115700+00:00" testType="13cdc9d9-ddb5-4fa4-a97d-d965ccfc6d4b" outcome="Passed" testListId="8c84fa94-04c1-424b-9868-57a2d4851a1d" relativeResultsDirectory="868217d3-4a8f-4036-9886-314c10b3f4b9" />
```
Comment (samholder) [inline test/dotnet-trx-merge.IntegrationTests/Fixtures/MergeWithMyFiles/_fv-az635-124_2023-10-04_11_32_15.trx:19]: if you search for this test result in the file: bd0f3e87-2d49-7223-2c00-e0924f9c83f5 you'll find that it is duplicated quite 8 times

File: test/dotnet-trx-merge.UnitTests/Fixtures/Merge/DuplicateTestIdAllPass.trx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (samholder) [inline test/dotnet-trx-merge.UnitTests/Fixtures/Merge/DuplicateTestIdAllPass.trx:1]: these are samples files I created from my real files. We need additional ones that have failures in them, so we can test the fix actually works for tests that failed in the first run and passed in the second run

**Comment by samholder** [comment]
I would like to understand what the expected behavior is if a test passed in an earlier run, but failed in a later run, what should the merge be doing?

File: test/dotnet-trx-merge.UnitTests/Fixtures/Merge/DuplicateTestIdAllPass.trx (hunk 4 above)
Comment (samholder) [reply test/dotnet-trx-merge.UnitTests/Fixtures/Merge/DuplicateTestIdAllPass.trx]: I've added the additional tests

**Comment by samholder** [comment]
> > I would like to understand what the expected behavior is if a test passed in an earlier run, but failed in a later run, what should the merge be doing?
> 
> Currently it is replacing with the failed one. I thought of improving the tool to have a flag to always keep the passed results but wanted to see if it was something that would make sense

ok this makes sense, given the use case is merging after a rerun of the failed tests

**Comment by ricardofslp** [comment]
Hi @samholder ,

The solution I think it works for this use case 

For some reason the build was not running that was why I changed the base branch.
Have you finish the work?

If yes I will merge and try to release it this afternoon.

**Review by ricardofslp — APPROVED** [review]

File: test/dotnet-trx-merge.IntegrationTests/DotNetTrxMergeTests.cs (hunk 1 above)
Comment (samholder) [reply test/dotnet-trx-merge.IntegrationTests/DotNetTrxMergeTests.cs:117]: @ricardofslp this want many to be merged before this was removed, as the test fails. I'll get a pr together to remove this. 

I only added this to show a real example of the data from the wild. I'm not at a computer now, but as soon as I can be ill get a pr together to remove this test and the large data file

Comment (ricardofslp) [reply test/dotnet-trx-merge.IntegrationTests/DotNetTrxMergeTests.cs:117]: No worries. I've removed in a PR I did before merging to main.
Please check after if something else should be removed and open a PR if it should

**Pull request merged**